### ISSUE TITLE
[IoT Workbench] Embedded Linux project Hangs using Docker Desktop preview with WSL2

### ISSUE THREAD

**Issue by darenm** [microsoft/vscode-iot-workbench#741]
I tried to create a project using docker desktop 2.1.1.0 from the edge channel configured to use WSL2.

![image](https://user-images.githubusercontent.com/1674843/64381983-b3ac6b00-cff1-11e9-9192-96f21c75fe77.png)

Unfortunately it hangs when trying to start the development container:

```log
Setting up container for folder or workspace: C:\Users\daren\Documents\IoTWorkbenchProjects\projects\ConnectionTest
Run: docker build -f C:\Users\daren\Documents\IoTWorkbenchProjects\projects\ConnectionTest\.devcontainer\Dockerfile -t rpi32_iothub C:\Users\daren\Documents\IoTWorkbenchProjects\projects\ConnectionTest\.devcontainer
Sending build context to Docker daemon  3.072kB
Step 1/1 : FROM mcr.microsoft.com/iot-device-cube:1.0.0-azure-c-app-20190705-ubuntu-arm32
 ---> 2de276df56b7
Successfully built 2de276df56b7
Successfully tagged rpi32_iothub:latest
SECURITY WARNING: You are building a Docker image from Windows against a non-Windows Docker host. All files and directories added to build context will have '-rwxr-xr-x' permissions. It is recommended to double check and reset permissions for sensitive files and directories.
Run: docker run -a STDOUT -a STDERR --mount type=bind,source=C:\Users\daren\Documents\IoTWorkbenchProjects\projects\ConnectionTest,target=/workspaces/ConnectionTest,consistency=cached -l vsch.quality=stable -l vsch.local.folder=C:\Users\daren\Documents\IoTWorkbenchProjects\projects\ConnectionTest -l vsch.remote.devPort=0 --entrypoint /bin/sh rpi32_iothub -c echo Container started ;  while sleep 1; do :; done
Container started
```
![image](https://user-images.githubusercontent.com/1674843/64381931-96779c80-cff1-11e9-8245-0abc6e3d06a6.png)

**Comment by zunhu** [comment]
Hi, hmm.. sometimes it is not hang and just not show up installing details.

if it is spend long time, you can close VSCode and re-open VSCode, Dev Container will be continue install.

**Comment by darenm** [comment]
I tried this multiple times, exiting VScode and restarted resulted in the same hang. However, as soon as I disabled the WSL2 feature of the Docker install, it worked. Therefore my issue still hangs - did you try it with the version of docker and WSL2 feature enabled?

https://docs.docker.com/docker-for-windows/wsl-tech-preview/

Thanks,

Daren
________________________________

**Comment by zunhu** [comment]
For WSL2 during our developing we haven't test on it.
I'm trying to use WSL2 to repo this issue, if so, I will mark this issue as a bug. Will update you soon.

**Comment by zunhu** [comment]
Hi darenm, I've tried to enable docker WSL2, but can't find WSL option from docker. Even I've installed WSL and ubuntu.
installed docker version: [Docker Desktop WSL 2 Tech Preview Installer](https://download.docker.com/win/edge/36883/Docker%20Desktop%20Installer.exe). 

May I know which version of docker have you installed? Or what I've missed.

![image](https://user-images.githubusercontent.com/40815873/64756990-51101d80-d563-11e9-99f6-db21bd50c6d2.png)
![image](https://user-images.githubusercontent.com/40815873/64757069-81f05280-d563-11e9-844f-2a38c8abe25f.png)

![image](https://user-images.githubusercontent.com/40815873/64756892-01315680-d563-11e9-946c-41d588a351dd.png)
![image](https://user-images.githubusercontent.com/40815873/64756953-29b95080-d563-11e9-9c07-5f9b9751bea9.png)

**Comment by zunhu** [comment]
After I installed win10 insider version 18980, I can enable wsl2 in docker.
Issue can be repro on my side and I will let developer know.

![image](https://user-images.githubusercontent.com/40815873/65010508-a7ee6c00-d942-11e9-8c26-196c7f3cd10f.png)

**Comment by zunhu** [comment]
Hi, since we just support local docker but not wsl1 wsl2 by design.
I mark this issue as a feature.

**Comment by darenm** [comment]
Are you guys aware that this will soon be the only way Docker Desktop runs on Windows? It will be replacing the HyperV implementation. See below.

[https://engineering.docker.com/2019/06/docker-hearts-wsl-2/](https://engineering.docker.com/2019/06/docker-hearts-wsl-2/)

**Comment by dilin-MS** [comment]
Hi, @darenm , could you please share your C drive in _Docker Desktop for WSL 2_ setting and re-try? Let me know if you still have problem. Thanks!
![dc1ffcc8-d8b7-412a-b87e-712d1e85038f](https://user-images.githubusercontent.com/41713563/65230079-d2d1ef00-daff-11e9-9d6b-d8ff420d7c35.jpg)

**Comment by darenm** [comment]
Yes it is shared. It looks like this is a broad issue with how VSCode interacts with Docker:

* https://github.com/microsoft/vscode-remote-release/issues/1390
* https://github.com/microsoft/vscode-remote-release/issues/1089
* https://github.com/microsoft/vscode-docker/issues/1199

**Comment by dilin-MS** [comment]
Yes, we would talk with VS Code Remote team to follow up this feature. Fix it ASAP. Thanks for your active participation!
